src/brainseg/create_fh_seg.py

```diff
@@ -0,0 +1,121 @@
+import os
+import sys
+import subprocess
+import nibabel as nib
+import numpy as np
+import scipy.ndimage
+import shutil
+
+def run_synthseg(conda_env, input_nifti, output_nifti, synthseg_script, synthseg_dir):
+    python_bin = os.path.join(conda_env, "bin", "python")
+    env = os.environ.copy()
+    env["PYTHONPATH"] = synthseg_dir
+
+    subprocess.run(
+        [python_bin, synthseg_script, "--i", input_nifti, "--o", output_nifti, "--robust"],
+        check=True, cwd=synthseg_dir, env=env
+    )
+
+def run_charm(conda_env, subject_id, input_nifti, charm_working_dir):
+    charm_bin = os.path.join(conda_env, "bin", "charm")
+    subject_dir = os.path.join(charm_working_dir, f"m2m_{subject_id}")
+
+    os.makedirs(charm_working_dir, exist_ok=True)
+
+    if os.path.exists(subject_dir):
+        shutil.rmtree(subject_dir)
+
+    subprocess.run(
+        [charm_bin, subject_id, input_nifti, "--forcesform", "--forcerun"],
+        check=True, cwd=charm_working_dir
+    )
+
+def isolate_labels(nifti_path, labels, output_nifti):
+    img = nib.load(nifti_path)
+    data = img.get_fdata().astype(np.int32)
```

Comment:
I found that you can do `img.get_fdata(dtype=....)` and that you can get the underlying Nifti data type with `img.get_data_type()`. This should be faster because it doesn't do any type conversion for label images of type integer.